minitask/new_import_data_v2.py

```diff
@@ -0,0 +1,162 @@
+            # used to store the element
+            element.clear()
+
+    total_number = len(data_set)
+    print('in total: ', total_number)
+    data_set_test = data_set[:total_number-2]
+    elastic_search = start_elastic_search()
+
+    if not elastic_search.indices.exists( index = 'wiki'):
+        # create the index
+	    elastic_search.indices.create(index = 'wiki')
+
+    for x in range(total_number-2):
+
+        print('adding: ', data_set_test[x]['title'])
+        index_elastic_search(data_set_test[x], elastic_search, x)
+
+        print(x,"/200000") 
+
+    search_index_test(elastic_search)
+
+
+    # elastic_search = es(["127.0.0.1"], timeout=35, max_retries=8, retry_on_timeout=True)
+    # .................
+
+    # pool = Pool(10)
+    # result_iter = pool.imap_unordered(process, data_set_test, chunksize=50)
+
+# def process(data):
+
+#     global elasticsearch
+
+#     print("thread_id :", multiprocessing.current_process())
+#     try:
+#         print('adding: ', data['title'])
+#         index_elastic_search(data, elastic_search)
+#     except (Exception) as e:
+#         print("Could not index document, {0}:".format(data["title"]), e) 
+
+
+# def index_elastic_search(data, elastic_search):
+#     # u6250082 Xuguang Song
+#     '''parameter: data, elastic search engine, ind'''
+    
+#     try:
+
+#         index_result = elastic_search.index(index='wiki', body=data)
+
+#     except:
+#         print(data)
+        
+#     print('success')
+
+
+def index_elastic_search(data, elastic_search, index):
+    # u6250082 Xuguang Song
+    '''parameter: data, elastic search engine, ind'''
+      
+    try:
+
+        index_result = elastic_search.index(index='news', body=data, id=index)
+
+    except:
+        print(data)
+    print('success')
+
+def start_elastic_search():
+    # u6250082 Xuguang Song
+    '''start elastic search'''
+
+    ip_url = ["127.0.0.1"]
+    # initialize elastic search
+    new_es = es(ip_url, timeout=35, max_retries=8, retry_on_timeout=True)
+    return new_es
+
+def search_index_test(elastic_search):
+    # u6250082 Xuguang Song
+    '''test with query to match ACT'''
+
+    q = "ACT"
+    position = "title"
+    test1 = {
+            "query": {
+                "match": {
+                    position: q
+                }
+            }
+        }
+    # output = elastic_search.search(index='news', doc_type='web_news', body=test1)
+    print('\n', 'searching for keyword: ', q, " ", "in", " article ", position, '\n')
+    output = elastic_search.search(index='news', body=test1)
+    print('searching finished with total time: ', output['took'], '\n')
+    print('result: ', '\n')
+    for hit in output['hits']['hits']:
+        # print search result
+        print ('match news title: ', hit['_source']['title'], '\n', 'match news link: ', hit['_source']['link'], '\n', '-------------------------------------------------------------')
+
+if __name__ == '__main__':
+    start_import() 
+    
```

Comment:
**[Required]** Missing a newline at the end of this file, probably has spaces in it.

Reply:
ok